### PR TITLE
Change `setTheme` type to `React.Dispatch<React.SetStateAction<string>>`.

### PR DESCRIPTION
**Pull request by yxshv**
The `setTheme()` function in `useTheme()` hook, Had a argument of type `string` which changed the `setThemeState` with the value passed in. But you couldn't send a function through `setTheme()` .

So, I have updated the `setTheme()` to support functions, through which previous values can be accessed and the state can be changed accordingly.

---

And I also, removed the `yarn` prefix in `test:e2e` command. Its not required and now it can be run with any package manager

### CODE REVIEW

**Comment by yxshv** [comment]
its been a month and no response :(

**Comment by pacocoursey** [comment]
@yxshv This is a very messy PR.

Keep your PRs to a single topic, in this case only include the setTheme changes. Please remove all code formatting and unrelated script changes. Remove the pnpm lockfile. Remove the random async IIFE.

Happy to review again when it's more focused and valuable.

**Comment by yxshv** [comment]
✔️ removed `pnpm-lock.yaml`
✔️ undone `scripts` changes
✔️ removed formattings
✔️ removed asynchronous function

**Review by pacocoursey — APPROVED** [review]

File: src/index.tsx [hunk 1]
```diff
@@ -81,15 +81,22 @@ const Theme: React.FC<ThemeProviderProps> = ({
...
-      try {
-        localStorage.setItem(storageKey, theme)
-      } catch (e) {
-        // Unsupported
-      }
+    (newTheme: React.SetStateAction<string>) => {
+
+      setThemeState(
+        prevTheme => {
+          const _theme = (newTheme instanceof Function) ? newTheme(prevTheme ?? defaultTheme) : prevTheme ?? defaultTheme;
```
Comment (B2o5T) [inline src/index.tsx:88]:
```suggestion
          const _theme = (typeof newTheme === 'function') ? newTheme(prevTheme ?? defaultTheme) : prevTheme ?? defaultTheme;
```

**Comment by yxshv** [comment]
anything else?

**Pull request closed without merging**